### PR TITLE
More pre commit hooks

### PR DESCRIPTION
**Pull request by jl-wynen**
Main part of #2906

This does not yet include `isort` as that would touch (almost) every python file and make the PR even harder to review. I'm going to add it later, but please review the current draft.

I ran `pre-commit run -a` and fixed all warnings. This is currently stricter than our CI. We can change the CI setup in a follow up PR.

### CODE REVIEW

**Comment by jl-wynen** [comment]
Turns out, the clang-format hook was out of date

File: pyproject.toml [hunk 1]
```diff
@@ -23,8 +23,23 @@ build-verbosity = 1
 [tool.cibuildwheel.macos]
 archs = ["x86_64", "arm64"]
 
+[tool.bandit]
+# Excluding tests because bandit doesn't like `assert`.
+exclude_dirs = ["docs/conf.py", "tests", "install", "conda/cmake-package-test", "tools", "setup.py"]
+
+[tool.codespell]
+ignore-words-list = "elemt"
```
Comment (SimonHeybrock) [inline pyproject.toml:31]: Where are we using this?

**Review by SimonHeybrock — APPROVED** [review]
Nice, lots of useful corrections!

File: pyproject.toml (hunk 1 above)
Comment (jl-wynen) [reply pyproject.toml:31]: In C++. There are template parameters called `ElemT`. Could rename them if you think ignoring this word is risky.

Comment (SimonHeybrock) [reply pyproject.toml:31]: Ok, so it is not case-sensitive?

You can leave it.

File: src/scipp/__init__.py [hunk 2]
```diff
@@ -54,39 +49,28 @@
 setattr(Variable, '__format__', _format_variable)
 del _format_variable
 
+from . import _binding
+from .compat.dict import from_dict, to_dict
+from .coords import show_graph, transform_coords
+# Mainly imported for docs
```
Comment (SimonHeybrock) [inline src/scipp/__init__.py:55]: What? Why?

File: src/scipp/__init__.py [hunk 3]
```diff
@@ -54,39 +49,28 @@
...
+    datetime, datetimes, density_to_counts, divide, dot, empty, empty_like, epoch, \
+    equal, erf, erfc, exp, flatten, floor, floor_divide, fold, full, full_like, \
+    geomspace, greater, greater_equal, group, groupby, hist, identical, index, \
+    isclose, isfinite, isinf, islinspace, isnan, isneginf, isposinf, issorted, less, \
+    less_equal, linspace, log, log10, logical_and, logical_not, logical_or, \
+    logical_xor, logspace, lookup, matrices, matrix, max, mean, merge, midpoints, min, \
+    mod, multiply, nan_to_num, nanhist, nanmax, nanmean, nanmin, nansum, negative, \
+    norm, not_equal, ones, ones_like, pow, rebin, reciprocal, round, scalar, sin, \
+    sort, sqrt, squeeze, stddevs, subtract, sum, tan, to, to_unit, transpose, values, \
+    variances, vector, vectors, where, zeros, zeros_like  # NOQA
```
Comment (SimonHeybrock) [inline src/scipp/__init__.py:68]: Hmm, I don't feel this is a good change. Having the imports split by file gave us a chance to see if something was missing, or adding it easily. Should we ignore the `__init__.py`?

Comment (jl-wynen) [reply src/scipp/__init__.py]: Excluded `__init__.py` and `core/__init__.py` because both have a bunch of rather special imports.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**